libs/filamat/src/GLSLPostProcessor.cpp

```diff
@@ -30,60 +27,63 @@
 #include "sca/GLSLTools.h"
 #include "shaders/MaterialInfo.h"
 
-#include <utils/Log.h>
 #include <filament/MaterialEnums.h>
 #include <private/filament/Variant.h>
 #include <private/filament/SibGenerator.h>
 
+#include <utils/Log.h>
+
+#include <sstream>
+#include <unordered_map>
+#include <vector>
+
 using namespace glslang;
 using namespace spirv_cross;
 using namespace spvtools;
+using namespace filament;
+using namespace filament::backend;
 
 namespace filamat {
 
-using BindingIndexMap = tsl::robin_map<std::string, uint16_t>;
+namespace msl {  // this is only used for MSL
+
+using BindingIndexMap = std::unordered_map<std::string, uint16_t>;
 
+UTILS_NOINLINE
 static void generateBindingIndexMap(const GLSLPostProcessor::Config& config,
-        filament::SamplerInterfaceBlock const& sib, BindingIndexMap& map) {
+        SamplerInterfaceBlock const& sib, BindingIndexMap& map) {
     const auto stageFlags = sib.getStageFlags();
-    if (!stageFlags.hasShaderType(config.shaderType)) {
-        return;
-    }
-    const auto& infoList = sib.getSamplerInfoList();
-    for (const auto& info : infoList) {
-        auto uniformName = filament::SamplerInterfaceBlock::getUniformName(
-                sib.getName().c_str(), info.name.c_str());
-        map[uniformName.c_str()] = map.size();
+    if (stageFlags.hasShaderType(config.shaderType)) {
```

Comment:
Is there a size advantage to inverting this if statement? I much prefer using a guard clause.

Reply:
oh not at all, I thought it was more readable :-)